### PR TITLE
archive: add TiDB v7.3

### PR DESCRIPTION
**Pull request by qiancai**
1. Update the docs.json the latest version
2. Add v7.3 to the homepage of the archive doc site.

### CODE REVIEW

File: markdown-pages/zh/tidb/_docHome.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ hide_leftNav: true
 
 | 版本        | 归档文档链接                                             | 归档日期 |
 | ---------- | ------------------------------------------------------ | ----------------- |
+| v7.3 (DMR) | [TiDB v7.3 文档](https://docs-archive.pingcap.com/zh/tidb/v7.3) | 2023 年 2 月 2 日 |
```
Comment (qiancai) [inline markdown-pages/zh/tidb/_docHome.md:23]:
```suggestion
| v7.3 (DMR) | [TiDB v7.3 文档](https://docs-archive.pingcap.com/zh/tidb/v7.3) | 2024 年 2 月 2 日 |
```

File: docs.json [hunk 2]
```diff
@@ -1,14 +1,15 @@
 {
   "docs": {
```
Comment (qiancai) [inline docs.json:2]: 这个文件是直接从 https://github.com/pingcap/docs-staging/pull/63 里复制替换过来的

File: docs.json
Comment (qiancai) [inline docs.json:1]: 这个文件是直接从 https://github.com/pingcap/docs-staging/pull/63 里复制替换过来的

**Review by Yuiham — APPROVED** [review]
lgtm

**Pull request merged**